Added color and opacity options

igormatos opened the pull request:
The color icon used is from Material Design Icons and the opacity is [this one](https://thenounproject.com/term/opacity/143231). And needs credit.

Commit:
Added color and opacity options

jpedroribeiro commented on chrome_extension/baseliner_v1.0.0.js:
```diff
@@ -11,7 +11,9 @@ Baseliner = {
-	baseline:		8,		//      ''
+	baseline:		12,		//      ''
+	opacity: 100,
+	baselinerColor: '#cccccc',
```
for readability, could you indent the lines above to match the ones line on line 12?

jpedroribeiro commented on chrome_extension/baseliner_v1.0.0.js:
```diff
@@ -30,6 +32,8 @@ Baseliner = {
 			this.baseline = this.getBaselineDataAttribute();
 			this.baselineTop = this.getTopDataAttribute();
+			this.baselinerColor = this.getColorDataAttribute();
+			this.opacity = this.getOpacityDataAttribute();
```
why not call it `baselineOpacity` instead of `opacity`? so it follows the pattern.

jpedroribeiro commented on chrome_extension/baseliner_v1.0.0.js:
```diff
@@ -98,23 +102,31 @@ Baseliner = {
 	 * Updates Baseliner with new values
 	 * @param newBaseline
 	 * @param newTop
+	 * @param newColor
```
you are also sending `newOpacity` so it should be on the docs comment like you did for `newColor`

jpedroribeiro commented on chrome_extension/baseliner_v1.0.0.js:
```diff
@@ -98,23 +102,31 @@ Baseliner = {
+	 * @param newColor
 	 */
-	update: function(newBaseline, newTop) {
+	// update: function(newBaseline, newTop) {
```
can we get rid of this comment line?

jpedroribeiro commented on chrome_extension/baseliner_v1.0.0.js:
```diff
@@ -137,9 +149,11 @@ Baseliner = {
 	 * @param baseline
 	 * @param top
 	 */
```
can you add the comments for the new params you added? (`opacity`, `color`)

jpedroribeiro commented on chrome_extension/baseliner_v1.0.0.js:
```diff
@@ -137,9 +149,11 @@ Baseliner = {
 	 * @param top
 	 */
-	setDataAttributes: function(baseline, top){
+	setDataAttributes: function(baseline, top, opacity, color){
```
if you are going to send the options (baseline, top, opacity, color) in different methods, can you always use the same order? `update()` is different than `setDataAttribute` and it can become easy to make mistakes in the future

jpedroribeiro commented on chrome_extension/color.all.min.js:
```diff
@@ -0,0 +1,5 @@
+
+!function(a,b){"use strict";function c(a,c,d,f,g){if("string"==typeof c){var c=v.txt2color(c);d=c.type,n[d]=c[d],g=g!==b?g:c.alpha}else if(c)for(var h in c)a[d][h]="Lab"===d?k(c[h],l[d][h][0],l[d][h][1]):k(c[h]/l[d][h][1],0,1);return g!==b&&(a.alpha=k(+g,0,1)),e(d,f?a:b)}function d(a,b,c){var d=m.options.grey,e={};return e.RGB={r:a.r,g:a.g,b:a.b},e.rgb={r:b.r,g:b.g,b:b.b},e.alpha=c,e.equivalentGr ...
+}function n(c){var d,e=c||a.event,f=e.which||e.keyCode,g=String.fromCharCode(f),h=document.activeElement,j=h.className.replace(bb.CSSPrefix,"").split("-"),k=j[0],l=j[1],m="alpha"===k,n="HEX"===k,o={k40:-1,k38:1,k34:-10,k33:10}["k"+f]/(m?100:1),p={HEX:/[0-9a-fA-F]/,Lab:/[\-0-9]/,alpha:/[\.0-9]/}[k]||/[0-9]/,q=S[k][k]||S[k][l],r=h.firstChild,s=H(h),u=r.data,w="0"!==u||n?u.split(""):[];return/^(?:27 ...
+//# sourceMappingURL=colorPicker.js.map
```
what is this file? can we remove it please.

jpedroribeiro commented on chrome_extension/jsColorPicker.min.js:
```diff
@@ -0,0 +1,5 @@
+
+!function(a,b){"use strict";function c(a,c,d,f,g){if("string"==typeof c){var c=v.txt2color(c);d=c.type,n[d]=c[d],g=g!==b?g:c.alpha}else if(c)for(var h in c)a[d][h]="Lab"===d?k(c[h],l[d][h][0],l[d][h][1]):k(c[h]/l[d][h][1],0,1);return g!==b&&(a.alpha=k(+g,0,1)),e(d,f?a:b)}function d(a,b,c){var d=m.options.grey,e={};return e.RGB={r:a.r,g:a.g,b:a.b},e.rgb={r:b.r,g:b.g,b:b.b},e.alpha=c,e.equivalentGr ...
+}function n(c){var d,e=c||a.event,f=e.which||e.keyCode,g=String.fromCharCode(f),h=document.activeElement,j=h.className.replace(bb.CSSPrefix,"").split("-"),k=j[0],l=j[1],m="alpha"===k,n="HEX"===k,o={k40:-1,k38:1,k34:-10,k33:10}["k"+f]/(m?100:1),p={HEX:/[0-9a-fA-F]/,Lab:/[\-0-9]/,alpha:/[\.0-9]/}[k]||/[0-9]/,q=S[k][k]||S[k][l],r=h.firstChild,s=H(h),u=r.data,w="0"!==u||n?u.split(""):[];return/^(?:27 ...
+//# sourceMappingURL=../colorPicker.js.map
```
can we remove this file?

jpedroribeiro commented on chrome_extension/jscolor-example.html:
```diff
@@ -0,0 +1,40 @@
+
+
+</body>
+</html>
```
can we remove this file?

jpedroribeiro commented on jscolor.min.js:
```diff
@@ -0,0 +1,10 @@
+ * @author  Jan Odvarko
+ *
+ * See usage examples at http://jscolor.com/examples/
+ */"use strict";window.jscolor||(window.jscolor=function(){var e={register:function(){e.attachDOMReadyEvent(e.init),e.attachEvent(document,"mousedown",e.onDocumentMouseDown),e.attachEvent(document,"touchstart",e.onDocumentTouchStart),e.attachEvent(window,"resize",e.onWindowResize)},init:function(){e.jscolor.lookupClass&&e.jscolor.installByClassName(e.jscolor.lookupClass)},tryInstallOnElements:fun ...
```
could we not have this file?

jpedroribeiro commented on chrome_extension/styles.css:
```diff
@@ -681,4 +687,13 @@ header {
+    margin-top: 3px;
+    border: none;
+    border-radius: 3px;
+    padding-left: 8px;
```
can you fix indentation here please

jpedroribeiro commented on chrome_extension/styles.css:
```diff
@@ -607,21 +607,27 @@ header {
         background-color: #f0f0f0;
     }
-
+  /*width: 100%;*/
```
can we remove this comment please

jpedroribeiro commented on chrome_extension/picker.js:
```diff
@@ -0,0 +1,8 @@
+      elm.style.backgroundColor = elm.value;
+      elm.style.color = colors.rgbaMixCustom.luminance > 0.22 ? '#222' : '#ddd';
+    }
+  });
```
do we need this file?

jpedroribeiro commented on chrome_extension/manifest.json:
```diff
@@ -16,5 +16,7 @@
\ No newline at end of file
+  ],
+  "web_accessible_resources": [
+    "line.svg", "main.js","jsColorPicker.min.js", "picker.js"]
```
is this necessary?

jpedroribeiro reviewed (changes requested):
Hi there igormatos,

Really like the updates proposed in this PR! Good job!
The idea of using SVG instead is smart and I wish I had thought about that sooner.

My only concern is the use of those JS files for the colour picker. I don't want this extension to have dependencies like that and it feels like it could be achieved just by using <input type=color> (https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color)

I've also made some comments more related about readability and code standards. 

What do you think?

igormatos commented:
@jpedroribeiro I will fix these things, thanks for the review. Some points I didn't pay attention. I am not a javascript developer, and some things were new to me hehe :p

Commit:
Source formatting

igormatos commented:
I've updated the PR. Indented the lines you asked, removed the some files and reordered the methods parameters. But the extension stills use the JS script to the color picker, i had some problem trying to use the input type="color", because when it opens the color picker window, it closes the extension tab, loosing the context, so i could't listen to onChange event. If you can make it work, I agree, it would be better, was my first time playing with chrome extension :))

jpedroribeiro commented:
Ah I see. Yeah, it makes perfect sense.

Would you mind giving me access to edit your PR branch? 
There are some updates I'd like to do before merging it into the master branch, like updating the Chrome Store  / website images. 

Here's some info if you need assistance: https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/

igormatos commented:
No problem. Edit allowed :)

igormatos commented:
You can remove the background image that was used too.

jpedroribeiro committed:
Code tidy up.

jpedroribeiro committed:
Moving js into main, tidy up on code and extension style

jpedroribeiro committed:
identation fix
